### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by Zegervg** [yudianzheng/SketchVideo#8]
Hi,

I know this is not really an issue, but can you give me an example prompt on how to run the soapbox. Since whenever I run it I dont get results that look like the one from the example in the github.

**Comment by Zegervg** [comment]
https://github.com/user-attachments/assets/47059ce1-fc41-43a3-a9c9-d3bb84d3e988

**Comment by Zegervg** [comment]
https://github.com/user-attachments/assets/6d1786ec-a4ac-493b-b601-7b5007eed3b7

**Comment by yudianzheng** [comment]
> init.mp4

what are your parameters? As the name of "init.mp4", I think you may just get the initialised strokes instead of optimised sketches.

**Comment by Zegervg** [comment]
Just the default parameters, but I don't
![Capture](https://github.com/user-attachments/assets/f218f4c5-de42-40b3-b2a5-eef02fb9b994)
 think its working either. The script appears to be running without errors but at the end it says 0% and just bails out.

**Comment by yudianzheng** [comment]
I believe the parameters are correct. The main issue is that the program terminates before optimization completes. It is intended to run for 2000 epochs, with an MP4 file output every 100 epochs

**Comment by Zegervg** [comment]
Yeah I see, do you know any solutions to this issue? Tried debugging it but without success.

**Comment by yudianzheng** [comment]
> Yeah I see, do you know any solutions to this issue? Tried debugging it but without success.

Logically speaking, such a thing shouldn't happen, and I haven't encountered this problem before either. You can set breakpoints to check why the process exits after initialization and at which step it exits.

**Comment by Zegervg** [comment]
Thanks for the advice, I've also tried CLIPasso with docker install which worked fine, could my gpu having to litlle amount of vram be part of the problem?

**Comment by yudianzheng** [comment]
> Thanks for the advice, I've also tried CLIPasso with docker install which worked fine, could my gpu having to litlle amount of vram be part of the problem?

Perhaps, you can try decreasing the number of strokes. It may help with low GPU resources.

**Comment by Zegervg** [comment]
quick question, what exact python version are you running this on and is this on linux or windows

**Comment by yudianzheng** [comment]
> quick question, what exact python version are you running this on and is this on linux or windows

python 3.7.5; Linux.

**Comment by Zegervg** [comment]
ubuntu?

**Comment by yudianzheng** [comment]
> ubuntu?

ubuntu20.04 cuda11.3 torch1.12.1.

**Comment by Zegervg** [comment]
ty

**Comment by Zegervg** [comment]
hi, me again. I lowered the values and now it seems to be working

"
python run_object_sketching.py --target_file $1 --focus foreground --data_folder ../data/dataset/$1 --consist_param 1.0 --frames_param 0.001 --num_strokes 4 --num_iter 101 --atlas_epoch '010000' --clip_model_name "RN101" --num_of_frames 49 --clip_RN_layer_weights "0.0,0.0,1.0,1.0,0.0" --clip_fc_loss_weight "0.1" --width "1.5" --device 0
"

![Capture](https://github.com/user-attachments/assets/4319e2bb-fb4c-4167-9d50-68c38bdce5fa)



Could you tell me if its even worth rendering with these bad parameters.

**Comment by yudianzheng** [comment]
I believe the main parameters influencing time are the number of strokes, number of frames, and number of iterations. However, the 3-hour runtime might be due to the capability of your GPU, as the requirements of these parameters are relatively low. Unfortunately, I cannot help further since I haven't fully optimized the code's efficiency. The consist_param and frames_param mainly influence the stability or flexibility of the video, while clip_RN_layer_weights and clip_fc_loss_weight influence the semantics. I advise you to only change consist_param, frames_param, or width, and use more strokes(16, 32, 48) and iterations(like 401).

**Comment by Zegervg** [comment]
Thanks for all the help, currently renting out an rtx 3090 to experiment with this <3

**Comment by Zegervg** [comment]
![Capture](https://github.com/user-attachments/assets/0da42c98-f439-4953-88b6-5ef2c0c6ded4)


Any clue what the reason of this issue could be?

**Comment by yudianzheng** [comment]
Has the program output the sketch videos yet? Is this feedback related to training the neural atlas or optimizing the sketch video?

**Comment by Zegervg** [comment]
It did not output any sketch videos. I Tried running this on my own dataset. Running the process_dataset and operate_atlas worked without any errors. But This happened when running the operate_clipavideo. It finished the first 1200 iterations and then had an error after the last 500 iterations.

**Comment by yudianzheng** [comment]
I haven't tried high iterations, like 1200, as that might cause some problems, such as exceeding the default maximum iterations. However, I think iterations higher than 800 will not make much difference. Some artifacts and poor results typically stem from the capabilities of the neural atlas and can be reduced by adjusting parameters like the number and width of strokes.
